draft-ietf-dprive-xfr-over-tls.md

```diff
@@ -364,37 +524,128 @@ The figure below provides an outline of the IXoT mechanism including NOTIFYs.
+Where any behavior is not explicitly described here, the behviour specified in
+[@RFC1995] MUST be followed. Any behaviour specified here takes precedence for
+AXoT implementations over that in [@RFC1995].
+
+### IXoT connection and message handling
+
+IXoT clients and servers SHOULD use EDNS0 Keepalive [RFC7828] to establish the
+connection timeouts to be used. The IXoT server may use the frequency of recent
+IXFRs to calculate an average update rate as input to the decision of what EDNS0
+Keepalive timeout to use. Note that the use of EDNS0 Keepalive enables IXoT
+servers to signal the desire to close a connection due to low resources by
+sending an EDNS0 Keepalive option with a timeout of 0. If the server does not
+support EDNS0 Keepalive the client MAY keep the connection open for a few
+seconds ([@RFC7766] recommends that servers use timeouts of at least a few
+seconds).
+
+In the original definition of IXFR, there arguably is an implicit assumption
+(probably unintentional) that a TCP connection is used for one and only one IXFR
+request. Indeed, several open source implementations currently take this
+approach. An IXoT server MUST be able to handle multiple IXFR requests on a
+single TLS connection (for the same and different zones), as well as to handle
+other query/response transactions over it.
+
+IXoT clients SHOULD re-use an existing open TLS connection when making any new
+IXoT request to the same primary, and for issuing SOA queries, instead of
+opening a new connection. (Non-IXoT traffic can also use an open TLS
+connection.) The number of TLS connections between a secondary and primary
+SHOULD be minimised.
+
+Valid reasons for not re-using existing connections might include:
+
+* reaching a configured limit for the number of outstanding queries allowed on a
+  single TLS connection
+* the message ID pool has already been exhausted on an open connection
+* a large number of timeouts or slow responses have occurred on an open
+  connection
+* an EDNS0 Keepalive option with a timeout of 0 has been received from the
+  server and the client is in the process of closing the connection
+
+If no TLS connections are currently open, IXoT clients MAY send SOA queries over
+UDP, TCP or TLS.
+
+An IXoT client SHOULD pipeline IXFR requests for different zones on a single TLS
+connection. An IXoT server SHOULD respond to those requests as soon as the
+response is available i.e. potentially out of order.
+
+### Condensation of responses
+
+QUESTION: Since this is a new specification should there be a requirement that
+IXoT servers are RECOMMENDED to condense responses as described in Section 6 of
+[RFC1995]. [RFC1995] document says this is optional and MAY be done. Whilst it
```

Comment:
Do we know if responses are usually condensed? We should try to limit barrier to adoption

Reply:
BIND does not condense, need to test other implementations. I doubt we can RECOMMEND this for the reason you give but it would be nice to push for it more...